.github/workflows/tests.yml

```diff
@@ -22,19 +22,65 @@ defaults:
     shell: bash -l {0}
 
 jobs:
-  runtime:
-    name: Runtime - ${{ matrix.os }}, Python ${{ matrix.python-version }}, Runtime ${{ matrix.runtime-version }}
+  tests:
+    name: Tests - ${{ matrix.category }} ${{ matrix.runtime-version }} ${{ matrix.os }} py${{ matrix.python-version }}
     runs-on: ${{ matrix.os }}
     timeout-minutes: 120
     strategy:
       fail-fast: false
       matrix:
-        os: ["ubuntu-latest"]
+        os: [ubuntu-latest]
         python-version: ["3.9"]
-        runtime-version: ["upstream", "latest", "0.0.4", "0.1.0"]
+        category: [runtime, benchmarks, stability]
```

Comment:
Even if you run all tests together, you'll still need a wordy `include` paragraph.

Instead of this:
```yaml
      matrix:
        os: [ubuntu-latest]
        python-version: ["3.9"]
        category: [runtime, benchmarks, stability]
        runtime-version: [upstream, latest, "0.0.4", "0.1.0"]
        include:
          # Run stability tests on Python 3.8
          - category: stability
            python-version: "3.8"
            runtime-version: upstream
            os: ubuntu-latest
       ...
```

it will look like this:
```yaml
      matrix:
        os: [ubuntu-latest]
        python-version: ["3.9"]
        test_dirs: [tests]
        runtime-version: [upstream, latest, "0.0.4", "0.1.0"]
        include:
          # Run stability tests on Python 3.8
          - test_dirs: tests/stability
            python-version: "3.8"
            runtime-version: upstream
            os: ubuntu-latest
         ...
```

In the next PR, I want to merge ab_tests.yaml into tests.yaml. In that PR I'll dynamically generate the whole matrix with `discover_ab_environments.py` (to be renamed); the matrix for non-A/B tests will be generated from parameters in `ci/config.yaml` (now `AB_environments/config.yaml`)

Reply:
Yeah, I was proposing just running the whole test suite on every matrix value. Perhaps overkill, however.

> In the next PR, I want to merge ab_tests.yaml into tests.yaml. In that PR I'll dynamically generate the whole matrix with `discover_ab_environments.py` (to be renamed); the matrix for non-A/B tests will be generated from parameters in `ci/config.yaml` (now `AB_environments/config.yaml`)

I'm a little concerned about the complexity of generating bespoke test matrices, and what it will mean for local testing. I was hoping to make the test matrix way *simpler*.